In [2]:
import pandas as pd
import datetime as dt
Real_Prop = pd.read_csv("../data/EXTR_RPSale.csv")
Res_Bldg = pd.read_csv("../data/EXTR_ResBldg.csv")
Parcel = pd.read_csv("../data/EXTR_Parcel.csv", encoding='latin-1')
codes = pd.read_csv('../data/EXTR_LookUp.csv')

/opt/anaconda3/envs/oy-env/lib/python3.8/site-packages/IPython/core/interactiveshell.py:3062: DtypeWarning: Columns (1,2) have mixed types.Specify dtype option on import or set low_memory=False.
  has_raised = await self.run_ast_nodes(code_ast.body, cell_name,
/opt/anaconda3/envs/oy-env/lib/python3.8/site-packages/IPython/core/interactiveshell.py:3062: DtypeWarning: Columns (11) have mixed types.Specify dtype option on import or set low_memory=False.
  has_raised = await self.run_ast_nodes(code_ast.body, cell_name,


Since we are only interested in 2019 data, using datetime to convert documentdate and filtering for 2019 sales

In [3]:
Real_Prop['DocumentDate'] = pd.to_datetime(Real_Prop['DocumentDate'])
Real_Prop19 = Real_Prop[Real_Prop['DocumentDate'].dt.year == 2019]
Real_Prop19

ExciseTaxNbr   Major Minor DocumentDate  SalePrice    RecordingNbr  \
4             3027422  213043   120   2019-12-20     560000  20191226000848   
118           2999169  919715   200   2019-07-08     192000  20190712001080   
144           3000673  894444   200   2019-06-26     185000  20190722001395   
164           3002257  940652   630   2019-07-22     435000  20190730001339   
445           2980836  937630   695   2019-03-28     550000  20190404001008   
...               ...     ...   ...          ...        ...             ...   
2089056       3004408   66000  2210   2019-08-07   41040000  20190812000849   
2089057       3004408   66000  2225   2019-08-07   41040000  20190812000849   
2089058       3004408   66000  2195   2019-08-07   41040000  20190812000849   
2089059       3004408   66000  2220   2019-08-07   41040000  20190812000849   
2089082       3009227  192300   170   2019-09-03    1550000  20190909000838   

        Volume Page PlatNbr PlatType  ... PropertyType PrincipalUse  \
4                                     ...           11            6   
118                                   ...            3            2   
144                                   ...            3            2   
164                                   ...           11            6   
445                                   ...            3            6   
...        ...  ...     ...      ...  ...          ...          ...   
2089056                               ...            3            7   
2089057                               ...            3            7   
2089058                               ...            3            7   
2089059                               ...            3            7   
2089082                               ...           11            6   

        SaleInstrument AFForestLand  AFCurrentUseLand  AFNonProfitUse  \
4                    3            N                 N               N   
118                  3            N                 N               N   
144                  3            N                 N               N   
164                  3            N                 N               N   
445                  3            N                 N               N   
...                ...          ...               ...             ...   
2089056             22            N                 N               N   
2089057             22            N                 N               N   
2089058             22            N                 N               N   
2089059             22            N                 N               N   
2089082              3            N                 N               N   

         AFHistoricProperty SaleReason PropertyClass SaleWarning  
4                         N          1             8              
118                       N          1             3              
144                       N          1             3              
164                       N          1             8              
445                       N          1             8              
...                     ...        ...           ...         ...  
2089056                   N         18             2          45  
2089057                   N         18             2          45  
2089058                   N         18             2          45  
2089059                   N         18             2          45  
2089082                   N          1             8              

[61363 rows x 24 columns]

formatting column names to make them more Pythonic!

In [4]:
Real_Prop19 = Real_Prop19.rename(columns=str.lower)

In [5]:
Real_Prop19

excisetaxnbr   major minor documentdate  saleprice    recordingnbr  \
4             3027422  213043   120   2019-12-20     560000  20191226000848   
118           2999169  919715   200   2019-07-08     192000  20190712001080   
144           3000673  894444   200   2019-06-26     185000  20190722001395   
164           3002257  940652   630   2019-07-22     435000  20190730001339   
445           2980836  937630   695   2019-03-28     550000  20190404001008   
...               ...     ...   ...          ...        ...             ...   
2089056       3004408   66000  2210   2019-08-07   41040000  20190812000849   
2089057       3004408   66000  2225   2019-08-07   41040000  20190812000849   
2089058       3004408   66000  2195   2019-08-07   41040000  20190812000849   
2089059       3004408   66000  2220   2019-08-07   41040000  20190812000849   
2089082       3009227  192300   170   2019-09-03    1550000  20190909000838   

        volume page platnbr plattype  ... propertytype principaluse  \
4                                     ...           11            6   
118                                   ...            3            2   
144                                   ...            3            2   
164                                   ...           11            6   
445                                   ...            3            6   
...        ...  ...     ...      ...  ...          ...          ...   
2089056                               ...            3            7   
2089057                               ...            3            7   
2089058                               ...            3            7   
2089059                               ...            3            7   
2089082                               ...           11            6   

        saleinstrument afforestland  afcurrentuseland  afnonprofituse  \
4                    3            N                 N               N   
118                  3            N                 N               N   
144                  3            N                 N               N   
164                  3            N                 N               N   
445                  3            N                 N               N   
...                ...          ...               ...             ...   
2089056             22            N                 N               N   
2089057             22            N                 N               N   
2089058             22            N                 N               N   
2089059             22            N                 N               N   
2089082              3            N                 N               N   

         afhistoricproperty salereason propertyclass salewarning  
4                         N          1             8              
118                       N          1             3              
144                       N          1             3              
164                       N          1             8              
445                       N          1             8              
...                     ...        ...           ...         ...  
2089056                   N         18             2          45  
2089057                   N         18             2          45  
2089058                   N         18             2          45  
2089059                   N         18             2          45  
2089082                   N          1             8              

[61363 rows x 24 columns]

padding major and minor and adding them together after converting them to string type

In [6]:
Real_Prop19['major'] = Real_Prop19['major'].astype(str)
Real_Prop19['minor'] = Real_Prop19['minor'].astype(str)

In [7]:
def major_padded(row):
    if len(row) == 6:
        return row
    elif len(row) == 5:
        return '0'+row
    elif len(row) == 4:
        return '00'+row
    elif len(row) == 3:
        return '000'+row
    elif len(row) == 2:
        return '0000'+row
    elif len(row) == 1:
        return '00000'+row
    else:
        return '000000'

def minor_padded(row):
    if len(row) == 4:
        return row
    elif len(row) == 3: 
        return '0'+row
    elif len(row) == 2:
        return '00'+row
    elif len(row) == 1:
        return '000'+row
    else:
        return '0000'
    
Real_Prop19['major'] = Real_Prop19.major.apply(major_padded)
Real_Prop19['minor'] = Real_Prop19.minor.apply(minor_padded)
Real_Prop19['major_minor'] = Real_Prop19['major']+Real_Prop19['minor']

In [8]:
Real_Prop19

excisetaxnbr   major minor documentdate  saleprice    recordingnbr  \
4             3027422  213043  0120   2019-12-20     560000  20191226000848   
118           2999169  919715  0200   2019-07-08     192000  20190712001080   
144           3000673  894444  0200   2019-06-26     185000  20190722001395   
164           3002257  940652  0630   2019-07-22     435000  20190730001339   
445           2980836  937630  0695   2019-03-28     550000  20190404001008   
...               ...     ...   ...          ...        ...             ...   
2089056       3004408  066000  2210   2019-08-07   41040000  20190812000849   
2089057       3004408  066000  2225   2019-08-07   41040000  20190812000849   
2089058       3004408  066000  2195   2019-08-07   41040000  20190812000849   
2089059       3004408  066000  2220   2019-08-07   41040000  20190812000849   
2089082       3009227  192300  0170   2019-09-03    1550000  20190909000838   

        volume page platnbr plattype  ... principaluse saleinstrument  \
4                                     ...            6              3   
118                                   ...            2              3   
144                                   ...            2              3   
164                                   ...            6              3   
445                                   ...            6              3   
...        ...  ...     ...      ...  ...          ...            ...   
2089056                               ...            7             22   
2089057                               ...            7             22   
2089058                               ...            7             22   
2089059                               ...            7             22   
2089082                               ...            6              3   

        afforestland afcurrentuseland  afnonprofituse  afhistoricproperty  \
4                  N                N               N                   N   
118                N                N               N                   N   
144                N                N               N                   N   
164                N                N               N                   N   
445                N                N               N                   N   
...              ...              ...             ...                 ...   
2089056            N                N               N                   N   
2089057            N                N               N                   N   
2089058            N                N               N                   N   
2089059            N                N               N                   N   
2089082            N                N               N                   N   

         salereason propertyclass salewarning major_minor  
4                 1             8              2130430120  
118               1             3              9197150200  
144               1             3              8944440200  
164               1             8              9406520630  
445               1             8              9376300695  
...             ...           ...         ...         ...  
2089056          18             2          45  0660002210  
2089057          18             2          45  0660002225  
2089058          18             2          45  0660002195  
2089059          18             2          45  0660002220  
2089082           1             8              1923000170  

[61363 rows x 25 columns]

moving the major_minor column to column1 position

In [9]:
col_name = 'major_minor'
last_col = Real_Prop19.pop(col_name)
Real_Prop19.insert(1, col_name, last_col)

In [10]:
Real_Prop19

excisetaxnbr major_minor   major minor documentdate  saleprice  \
4             3027422  2130430120  213043  0120   2019-12-20     560000   
118           2999169  9197150200  919715  0200   2019-07-08     192000   
144           3000673  8944440200  894444  0200   2019-06-26     185000   
164           3002257  9406520630  940652  0630   2019-07-22     435000   
445           2980836  9376300695  937630  0695   2019-03-28     550000   
...               ...         ...     ...   ...          ...        ...   
2089056       3004408  0660002210  066000  2210   2019-08-07   41040000   
2089057       3004408  0660002225  066000  2225   2019-08-07   41040000   
2089058       3004408  0660002195  066000  2195   2019-08-07   41040000   
2089059       3004408  0660002220  066000  2220   2019-08-07   41040000   
2089082       3009227  1923000170  192300  0170   2019-09-03    1550000   

           recordingnbr volume page platnbr  ... propertytype principaluse  \
4        20191226000848                      ...           11            6   
118      20190712001080                      ...            3            2   
144      20190722001395                      ...            3            2   
164      20190730001339                      ...           11            6   
445      20190404001008                      ...            3            6   
...                 ...    ...  ...     ...  ...          ...          ...   
2089056  20190812000849                      ...            3            7   
2089057  20190812000849                      ...            3            7   
2089058  20190812000849                      ...            3            7   
2089059  20190812000849                      ...            3            7   
2089082  20190909000838                      ...           11            6   

        saleinstrument afforestland afcurrentuseland  afnonprofituse  \
4                    3            N                N               N   
118                  3            N                N               N   
144                  3            N                N               N   
164                  3            N                N               N   
445                  3            N                N               N   
...                ...          ...              ...             ...   
2089056             22            N                N               N   
2089057             22            N                N               N   
2089058             22            N                N               N   
2089059             22            N                N               N   
2089082              3            N                N               N   

         afhistoricproperty  salereason propertyclass salewarning  
4                         N           1             8              
118                       N           1             3              
144                       N           1             3              
164                       N           1             8              
445                       N           1             8              
...                     ...         ...           ...         ...  
2089056                   N          18             2          45  
2089057                   N          18             2          45  
2089058                   N          18             2          45  
2089059                   N          18             2          45  
2089082                   N           1             8              

[61363 rows x 25 columns]

investigating propertytype to see the unique entries

In [11]:
Real_Prop19.propertytype.unique()

array([11,  3, 14, 51,  1, 10,  2, 59, 91, 16,  4, 12,  5,  0, 13, 58, 69,
       50, 99, 63, 18, 45, 79, 83, 65, 94,  6, 53, 68, 44, 54, 39, 85, 64,
       86, 55, 47, 19, 88, 17, 33, 15, 96, 27, 23, 46, 48, 52, 61, 81,  9,
       66, 21, 74])

The only propertytypes we are interested in are:

11 - Household, single family units

12 - Multiple family residence (Residential, 2-4 units)

14 - Residential condominiums

18 - All other residential not elsewhere coded

19 - Vacation and cabin (decided to not include vacation homes, probably first time homebuyers don't really go for these)

In [12]:
Real_Prop19 = Real_Prop19[Real_Prop19['propertytype'].isin([11, 12, 14, 18])]
Real_Prop19

excisetaxnbr major_minor   major minor documentdate  saleprice  \
4             3027422  2130430120  213043  0120   2019-12-20     560000   
164           3002257  9406520630  940652  0630   2019-07-22     435000   
456           2970575  8669201280  866920  1280   2019-01-16          0   
497           3025919  0565250340  056525  0340   2019-12-12     311000   
517           2975376  0984000450  098400  0450   2019-02-20     409950   
...               ...         ...     ...   ...          ...        ...   
2088838       2986110  8732340080  873234  0080   2019-04-10    2375000   
2088923       3012006  3904970150  390497  0150   2019-09-18     850000   
2088953       2974971  2460900090  246090  0090   2019-02-13     686000   
2088987       2998779  1438000420  143800  0420   2019-07-09     695000   
2089082       3009227  1923000170  192300  0170   2019-09-03    1550000   

           recordingnbr volume page platnbr  ... propertytype principaluse  \
4        20191226000848                      ...           11            6   
164      20190730001339                      ...           11            6   
456      20190118000160                      ...           14            2   
497      20191217001069                      ...           14            2   
517      20190228000242                      ...           11            6   
...                 ...    ...  ...     ...  ...          ...          ...   
2088838  20190503001267                      ...           14            2   
2088923  20190925001198                      ...           11            6   
2088953  20190226000416                      ...           11            6   
2088987  20190711000444                      ...           11            6   
2089082  20190909000838                      ...           11            6   

        saleinstrument afforestland afcurrentuseland  afnonprofituse  \
4                    3            N                N               N   
164                  3            N                N               N   
456                  3            N                N               N   
497                  3            N                N               N   
517                  3            N                N               N   
...                ...          ...              ...             ...   
2088838              3            N                N               N   
2088923              3            N                N               N   
2088953              3            N                N               N   
2088987              3            N                N               N   
2089082              3            N                N               N   

         afhistoricproperty  salereason propertyclass salewarning  
4                         N           1             8              
164                       N           1             8              
456                       N           1             3              
497                       N           1             3              
517                       N           1             8              
...                     ...         ...           ...         ...  
2088838                   N           1             3              
2088923                   N           1             8              
2088953                   N           1             8              
2088987                   N           1             8              
2089082                   N           1             8              

[30516 rows x 25 columns]

This has shrunk our data from 61363 entries to 30516

Let's only keep entries that have a Saleprice of more than 0

In [13]:
Real_Prop19 = Real_Prop19[Real_Prop19['saleprice'] > 0]
Real_Prop19

excisetaxnbr major_minor   major minor documentdate  saleprice  \
4             3027422  2130430120  213043  0120   2019-12-20     560000   
164           3002257  9406520630  940652  0630   2019-07-22     435000   
497           3025919  0565250340  056525  0340   2019-12-12     311000   
517           2975376  0984000450  098400  0450   2019-02-20     409950   
628           2976405  0302000645  030200  0645   2019-03-04     380000   
...               ...         ...     ...   ...          ...        ...   
2088838       2986110  8732340080  873234  0080   2019-04-10    2375000   
2088923       3012006  3904970150  390497  0150   2019-09-18     850000   
2088953       2974971  2460900090  246090  0090   2019-02-13     686000   
2088987       2998779  1438000420  143800  0420   2019-07-09     695000   
2089082       3009227  1923000170  192300  0170   2019-09-03    1550000   

           recordingnbr volume page platnbr  ... propertytype principaluse  \
4        20191226000848                      ...           11            6   
164      20190730001339                      ...           11            6   
497      20191217001069                      ...           14            2   
517      20190228000242                      ...           11            6   
628      20190306000786                      ...           11            6   
...                 ...    ...  ...     ...  ...          ...          ...   
2088838  20190503001267                      ...           14            2   
2088923  20190925001198                      ...           11            6   
2088953  20190226000416                      ...           11            6   
2088987  20190711000444                      ...           11            6   
2089082  20190909000838                      ...           11            6   

        saleinstrument afforestland afcurrentuseland  afnonprofituse  \
4                    3            N                N               N   
164                  3            N                N               N   
497                  3            N                N               N   
517                  3            N                N               N   
628                  3            N                N               N   
...                ...          ...              ...             ...   
2088838              3            N                N               N   
2088923              3            N                N               N   
2088953              3            N                N               N   
2088987              3            N                N               N   
2089082              3            N                N               N   

         afhistoricproperty  salereason propertyclass salewarning  
4                         N           1             8              
164                       N           1             8              
497                       N           1             3              
517                       N           1             8              
628                       N           1             8          10  
...                     ...         ...           ...         ...  
2088838                   N           1             3              
2088923                   N           1             8              
2088953                   N           1             8              
2088987                   N           1             8              
2089082                   N           1             8              

[21372 rows x 25 columns]

In [14]:
Real_Prop19.saleprice.agg(['min', 'max', 'std', 'mean'])

min     1.000000e+01
max     5.300000e+07
std     6.747552e+05
mean    7.585140e+05
Name: saleprice, dtype: float64

we now have 21372 entries, checking to see the principal use case for these entries 

In [15]:
Real_Prop19['principaluse'].unique()

array([6, 2, 7])

6 - Residential

2 - Condominium

7 - Commercial

Lets remove entries relating to commercial use

In [16]:
Real_Prop19 = Real_Prop19[Real_Prop19['principaluse'].isin([6,2])]
Real_Prop19

excisetaxnbr major_minor   major minor documentdate  saleprice  \
4             3027422  2130430120  213043  0120   2019-12-20     560000   
164           3002257  9406520630  940652  0630   2019-07-22     435000   
497           3025919  0565250340  056525  0340   2019-12-12     311000   
517           2975376  0984000450  098400  0450   2019-02-20     409950   
628           2976405  0302000645  030200  0645   2019-03-04     380000   
...               ...         ...     ...   ...          ...        ...   
2088838       2986110  8732340080  873234  0080   2019-04-10    2375000   
2088923       3012006  3904970150  390497  0150   2019-09-18     850000   
2088953       2974971  2460900090  246090  0090   2019-02-13     686000   
2088987       2998779  1438000420  143800  0420   2019-07-09     695000   
2089082       3009227  1923000170  192300  0170   2019-09-03    1550000   

           recordingnbr volume page platnbr  ... propertytype principaluse  \
4        20191226000848                      ...           11            6   
164      20190730001339                      ...           11            6   
497      20191217001069                      ...           14            2   
517      20190228000242                      ...           11            6   
628      20190306000786                      ...           11            6   
...                 ...    ...  ...     ...  ...          ...          ...   
2088838  20190503001267                      ...           14            2   
2088923  20190925001198                      ...           11            6   
2088953  20190226000416                      ...           11            6   
2088987  20190711000444                      ...           11            6   
2089082  20190909000838                      ...           11            6   

        saleinstrument afforestland afcurrentuseland  afnonprofituse  \
4                    3            N                N               N   
164                  3            N                N               N   
497                  3            N                N               N   
517                  3            N                N               N   
628                  3            N                N               N   
...                ...          ...              ...             ...   
2088838              3            N                N               N   
2088923              3            N                N               N   
2088953              3            N                N               N   
2088987              3            N                N               N   
2089082              3            N                N               N   

         afhistoricproperty  salereason propertyclass salewarning  
4                         N           1             8              
164                       N           1             8              
497                       N           1             3              
517                       N           1             8              
628                       N           1             8          10  
...                     ...         ...           ...         ...  
2088838                   N           1             3              
2088923                   N           1             8              
2088953                   N           1             8              
2088987                   N           1             8              
2089082                   N           1             8              

[21260 rows x 25 columns]

down to 21260 entries. lets give one last check for property class

In [17]:
Real_Prop19['propertyclass'].unique()

array([8, 3, 7, 9])

7 - Res-Land only, we can remove that from our dataframe to only reflect residential property

keeping only:

8 - Residential- Improved Property

3 - Condominium

9 - Res/Condo mobile

In [18]:
Real_Prop19 = Real_Prop19[Real_Prop19['propertyclass'].isin([8,3,9])]
Real_Prop19

excisetaxnbr major_minor   major minor documentdate  saleprice  \
4             3027422  2130430120  213043  0120   2019-12-20     560000   
164           3002257  9406520630  940652  0630   2019-07-22     435000   
497           3025919  0565250340  056525  0340   2019-12-12     311000   
517           2975376  0984000450  098400  0450   2019-02-20     409950   
628           2976405  0302000645  030200  0645   2019-03-04     380000   
...               ...         ...     ...   ...          ...        ...   
2088838       2986110  8732340080  873234  0080   2019-04-10    2375000   
2088923       3012006  3904970150  390497  0150   2019-09-18     850000   
2088953       2974971  2460900090  246090  0090   2019-02-13     686000   
2088987       2998779  1438000420  143800  0420   2019-07-09     695000   
2089082       3009227  1923000170  192300  0170   2019-09-03    1550000   

           recordingnbr volume page platnbr  ... propertytype principaluse  \
4        20191226000848                      ...           11            6   
164      20190730001339                      ...           11            6   
497      20191217001069                      ...           14            2   
517      20190228000242                      ...           11            6   
628      20190306000786                      ...           11            6   
...                 ...    ...  ...     ...  ...          ...          ...   
2088838  20190503001267                      ...           14            2   
2088923  20190925001198                      ...           11            6   
2088953  20190226000416                      ...           11            6   
2088987  20190711000444                      ...           11            6   
2089082  20190909000838                      ...           11            6   

        saleinstrument afforestland afcurrentuseland  afnonprofituse  \
4                    3            N                N               N   
164                  3            N                N               N   
497                  3            N                N               N   
517                  3            N                N               N   
628                  3            N                N               N   
...                ...          ...              ...             ...   
2088838              3            N                N               N   
2088923              3            N                N               N   
2088953              3            N                N               N   
2088987              3            N                N               N   
2089082              3            N                N               N   

         afhistoricproperty  salereason propertyclass salewarning  
4                         N           1             8              
164                       N           1             8              
497                       N           1             3              
517                       N           1             8              
628                       N           1             8          10  
...                     ...         ...           ...         ...  
2088838                   N           1             3              
2088923                   N           1             8              
2088953                   N           1             8              
2088987                   N           1             8              
2089082                   N           1             8              

[20999 rows x 25 columns]

investigated salewarnings and decided to skip and power through the rest.....

In [19]:
Real_Prop19.salewarning.value_counts()

            19598
15            281
26            214
40            184
41            152
            ...  
10 15 46        1
13 26           1
12 15 51        1
31              1
23 51           1
Name: salewarning, Length: 98, dtype: int64

Now checking to see if major_minor are unique or not

In [20]:
len(Real_Prop19) - len(Real_Prop19.major_minor.unique())

377

moving along and not worrying about the major_minor repeats, since it could be due to successive sale of the same property in 2019 or other reasons... can explore later..... (flash forwrd -- decided to drop the repeats here)

In [21]:
Real_Prop19['major_minor'].value_counts() > 1


8843900445     True
3122069048     True
1865820170     True
1972202382     True
8847500000     True
              ...  
2867300300    False
2770605165    False
3286800160    False
2874600035    False
8707100015    False
Name: major_minor, Length: 20622, dtype: bool

sorting the dataframe based on the document date

In [22]:
Real_Prop19 = Real_Prop19.sort_values(by='documentdate')
Real_Prop19

excisetaxnbr major_minor   major minor documentdate  saleprice  \
191874        2968945  4077800202  407780  0202   2019-01-01    1910000   
740217        2969148  1279001710  127900  1710   2019-01-01     135000   
364451        2970528  6190530110  619053  0110   2019-01-01    1089950   
1055892       2969165  0425400040  042540  0040   2019-01-01     412500   
959454        2970941  3629910780  362991  0780   2019-01-01     820000   
...               ...         ...     ...   ...          ...        ...   
2020279       3029940  9514050600  951405  0600   2019-12-31     921000   
1661931       3029386  2473350830  247335  0830   2019-12-31     627000   
2084034       3029011  5528700230  552870  0230   2019-12-31     870000   
1598521       3028733  7694210140  769421  0140   2019-12-31     640000   
510020        3029042  8722100825  872210  0825   2019-12-31    1300000   

           recordingnbr volume page platnbr  ... propertytype principaluse  \
191874   20190103000654                      ...           11            6   
740217   20190104000880                      ...           14            2   
364451   20190117000670                      ...           11            6   
1055892  20190104000953                      ...           11            6   
959454   20190122001427                      ...           11            6   
...                 ...    ...  ...     ...  ...          ...          ...   
2020279  20200115000600                      ...           11            6   
1661931  20200109001152                      ...           11            6   
2084034  20200107000738                      ...           14            2   
1598521  20200103000855                      ...           11            6   
510020   20200107001115                      ...           11            6   

        saleinstrument afforestland afcurrentuseland  afnonprofituse  \
191874               3            N                N               N   
740217               3            N                N               N   
364451               3            N                N               N   
1055892              3            N                N               N   
959454               3            N                N               N   
...                ...          ...              ...             ...   
2020279              3            N                N               N   
1661931              3            N                N               N   
2084034              3            N                N               N   
1598521              3            N                N               N   
510020               3            N                N               N   

         afhistoricproperty  salereason propertyclass salewarning  
191874                    N           1             8              
740217                    N           1             3              
364451                    N           1             8              
1055892                   N           1             8              
959454                    N           1             8          40  
...                     ...         ...           ...         ...  
2020279                   N           1             8              
1661931                   N           1             8              
2084034                   N           1             3              
1598521                   N           1             8              
510020                    N           1             8              

[20999 rows x 25 columns]

looking into the repeated major_minors to see any trend

In [23]:
Real_Prop19[Real_Prop19['major_minor'].isin(['8843900445','8847500000'])]


excisetaxnbr major_minor   major minor documentdate  saleprice  \
901520        2975141  8843900445  884390  0445   2019-02-14     699900   
249613        2975737  8843900445  884390  0445   2019-02-14     815000   
1858266       2975726  8843900445  884390  0445   2019-02-14     776900   
1748638       2975973  8843900445  884390  0445   2019-02-15     782900   
1982370       2976878  8847500000  884750  0000   2019-03-01     784684   
1181970       2976965  8843900445  884390  0445   2019-03-01     706900   
1738380       2980974  8843900445  884390  0445   2019-03-21     799900   
967675        2993692  8847500000  884750  0000   2019-06-11     764990   
1260009       3020478  8847500000  884750  0000   2019-11-05     698654   

           recordingnbr volume page platnbr  ... propertytype principaluse  \
901520   20190226001189                      ...           11            6   
249613   20190228001472                      ...           11            6   
1858266  20190228001436                      ...           11            6   
1748638  20190301000951                      ...           11            6   
1982370  20190311000330                      ...           14            2   
1181970  20190311000874                      ...           11            6   
1738380  20190405000497                      ...           11            6   
967675   20190614000861                      ...           14            2   
1260009  20191114000503                      ...           14            2   

        saleinstrument afforestland afcurrentuseland  afnonprofituse  \
901520               3            N                N               N   
249613               3            N                N               N   
1858266              3            N                N               N   
1748638              3            N                N               N   
1982370              3            N                N               N   
1181970              3            N                N               N   
1738380              3            N                N               N   
967675               3            N                N               N   
1260009              3            N                N               N   

         afhistoricproperty  salereason propertyclass salewarning  
901520                    N           1             8              
249613                    N           1             8              
1858266                   N           1             8              
1748638                   N           1             8              
1982370                   N           1             3          43  
1181970                   N           1             8              
1738380                   N           1             8              
967675                    N           1             3              
1260009                   N           1             3              

[9 rows x 25 columns]

Lets drop the blank columns 'volume', 'page', 'platnbr', 'plattype', 'platlot', 'platblock', 'sellername', 'buyername'

In [24]:
Real_Prop19.drop_duplicates(subset='major_minor', keep='last', inplace=True)

In [25]:
cols_to_drop = ['volume', 'page', 'platnbr', 'plattype', 'platlot', 'platblock', 'sellername', 'buyername']
Real_Prop19.drop(columns=cols_to_drop, inplace=True)

In [26]:
Real_Prop19

excisetaxnbr major_minor   major minor documentdate  saleprice  \
191874        2968945  4077800202  407780  0202   2019-01-01    1910000   
740217        2969148  1279001710  127900  1710   2019-01-01     135000   
364451        2970528  6190530110  619053  0110   2019-01-01    1089950   
1055892       2969165  0425400040  042540  0040   2019-01-01     412500   
1578500       2990693  2607870010  260787  0010   2019-01-01     400000   
...               ...         ...     ...   ...          ...        ...   
2020279       3029940  9514050600  951405  0600   2019-12-31     921000   
1661931       3029386  2473350830  247335  0830   2019-12-31     627000   
2084034       3029011  5528700230  552870  0230   2019-12-31     870000   
1598521       3028733  7694210140  769421  0140   2019-12-31     640000   
510020        3029042  8722100825  872210  0825   2019-12-31    1300000   

           recordingnbr  propertytype  principaluse  saleinstrument  \
191874   20190103000654            11             6               3   
740217   20190104000880            14             2               3   
364451   20190117000670            11             6               3   
1055892  20190104000953            11             6               3   
1578500                            11             2               3   
...                 ...           ...           ...             ...   
2020279  20200115000600            11             6               3   
1661931  20200109001152            11             6               3   
2084034  20200107000738            14             2               3   
1598521  20200103000855            11             6               3   
510020   20200107001115            11             6               3   

        afforestland afcurrentuseland afnonprofituse afhistoricproperty  \
191874             N                N              N                  N   
740217             N                N              N                  N   
364451             N                N              N                  N   
1055892            N                N              N                  N   
1578500            N                N              N                  N   
...              ...              ...            ...                ...   
2020279            N                N              N                  N   
1661931            N                N              N                  N   
2084034            N                N              N                  N   
1598521            N                N              N                  N   
510020             N                N              N                  N   

         salereason  propertyclass salewarning  
191874            1              8              
740217            1              3              
364451            1              8              
1055892           1              8              
1578500           1              3              
...             ...            ...         ...  
2020279           1              8              
1661931           1              8              
2084034           1              3              
1598521           1              8              
510020            1              8              

[20622 rows x 17 columns]

At present we have 20999 entries that satisfy all our conditions for residential propert sales in King county in 2019

Now bringing in the Residential Buildings table to mearge with our current datframe for features of the properties we have narrowed down.

We'll first clean up column names, pad up major and minor and combine them.

In [27]:
Res_Bldg.rename(columns=str.lower, inplace=True)

In [28]:
Res_Bldg['major'] = Res_Bldg['major'].astype(str)
Res_Bldg['minor'] = Res_Bldg['minor'].astype(str)

In [29]:
Res_Bldg['major'] = Res_Bldg['major'].apply(major_padded)
Res_Bldg['minor'] = Res_Bldg['minor'].apply(minor_padded)

In [30]:
Res_Bldg['major_minor'] = Res_Bldg['major']+Res_Bldg['minor']

In [31]:
col_name = 'major_minor'
last_col = Res_Bldg.pop(col_name)
Res_Bldg.insert(0, col_name, last_col)

In [32]:
Res_Bldg.head(2)

major_minor   major minor  bldgnbr  nbrlivingunits  \
0  0461000935  046100  0935        1               1   
1  0461001165  046100  1165        1               1   

                     address buildingnumber fraction directionprefix  \
0  7349   10TH AVE NW  98117          7349                             
1  7342   11TH AVE NW  98117          7342                             

                  streetname  ... fpmultistory fpfreestanding fpadditional  \
0  10TH                       ...            0              0            0   
1  11TH                       ...            0              0            0   

   yrbuilt  yrrenovated  pcntcomplete  obsolescence  pcntnetcondition  \
0     1927            0             0             0                 0   
1     2019            0            53             0                 0   

   condition  addnlcost  
0          3          0  
1          3          0  

[2 rows x 51 columns]

looks like residential buildings don't have condominiums (Minor=0000), we'll have to be prepared to loose more sale data :(

In [33]:
sum(Res_Bldg.minor == '0000')

0

In [34]:
Res_Bldg.loc[1]

major_minor                          0461001165
major                                    046100
minor                                      1165
bldgnbr                                       1
nbrlivingunits                                1
address               7342   11TH AVE NW  98117
buildingnumber                            7342 
fraction                                       
directionprefix                                
streetname            11TH                     
streettype                               AVE   
directionsuffix                              NW
zipcode                                   98117
stories                                       2
bldggrade                                     8
bldggradevar                                  0
sqft1stfloor                               1600
sqfthalffloor                                 0
sqft2ndfloor                               1560
sqftupperfloor                                0
sqftunfinfull                           

In [35]:
Res_Bldg.columns

Index(['major_minor', 'major', 'minor', 'bldgnbr', 'nbrlivingunits', 'address',
       'buildingnumber', 'fraction', 'directionprefix', 'streetname',
       'streettype', 'directionsuffix', 'zipcode', 'stories', 'bldggrade',
       'bldggradevar', 'sqft1stfloor', 'sqfthalffloor', 'sqft2ndfloor',
       'sqftupperfloor', 'sqftunfinfull', 'sqftunfinhalf', 'sqfttotliving',
       'sqfttotbasement', 'sqftfinbasement', 'finbasementgrade',
       'sqftgaragebasement', 'sqftgarageattached', 'daylightbasement',
       'sqftopenporch', 'sqftenclosedporch', 'sqftdeck', 'heatsystem',
       'heatsource', 'brickstone', 'viewutilization', 'bedrooms',
       'bathhalfcount', 'bath3qtrcount', 'bathfullcount', 'fpsinglestory',
       'fpmultistory', 'fpfreestanding', 'fpadditional', 'yrbuilt',
       'yrrenovated', 'pcntcomplete', 'obsolescence', 'pcntnetcondition',
       'condition', 'addnlcost'],
      dtype='object')

merging together Property sales in 2019 `Real_Prop19` with `Res_Bldg` purely based on their `major_minor`

In [36]:
prop_res = Real_Prop19.merge(Res_Bldg, on='major_minor', suffixes=("", "_resbldg"))

In [37]:
prop_res.describe()

excisetaxnbr     saleprice  propertytype  principaluse  saleinstrument  \
count  1.842000e+04  1.842000e+04  18420.000000       18420.0    18420.000000   
mean   2.999495e+06  7.965495e+05     11.027904           6.0        3.080565   
std    1.664883e+04  5.918905e+05      0.310664           0.0        1.221548   
min    2.968844e+06  1.000000e+01     11.000000           6.0        2.000000   
25%    2.985139e+06  4.700000e+05     11.000000           6.0        3.000000   
50%    3.000280e+06  6.570000e+05     11.000000           6.0        3.000000   
75%    3.012881e+06  9.070000e+05     11.000000           6.0        3.000000   
max    3.050564e+06  1.605000e+07     18.000000           6.0       26.000000   

         salereason  propertyclass       bldgnbr  nbrlivingunits  \
count  18420.000000        18420.0  18420.000000    18420.000000   
mean       1.001846            8.0      1.007709        1.014767   
std        0.177136            0.0      0.101272        0.140188   
min        1.000000            8.0      1.000000        1.000000   
25%        1.000000            8.0      1.000000        1.000000   
50%        1.000000            8.0      1.000000        1.000000   
75%        1.000000            8.0      1.000000        1.000000   
max       18.000000            8.0      7.000000        3.000000   

            stories  ...  fpmultistory  fpfreestanding  fpadditional  \
count  18420.000000  ...  18420.000000    18420.000000  18420.000000   
mean       1.517101  ...      0.303420        0.068187      0.175353   
std        0.534372  ...      0.500241        0.260962      0.398815   
min        1.000000  ...      0.000000        0.000000      0.000000   
25%        1.000000  ...      0.000000        0.000000      0.000000   
50%        1.500000  ...      0.000000        0.000000      0.000000   
75%        2.000000  ...      1.000000        0.000000      0.000000   
max        3.500000  ...      5.000000        2.000000      3.000000   

            yrbuilt   yrrenovated  pcntcomplete  obsolescence  \
count  18420.000000  18420.000000  18420.000000  18420.000000   
mean    1974.662486    100.657763      0.267101      0.102497   
std       30.389759    437.288542      4.646596      2.222057   
min     1900.000000      0.000000      0.000000      0.000000   
25%     1954.000000      0.000000      0.000000      0.000000   
50%     1978.000000      0.000000      0.000000      0.000000   
75%     2001.000000      0.000000      0.000000      0.000000   
max     2020.000000   2020.000000    100.000000     99.000000   

       pcntnetcondition     condition     addnlcost  
count      18420.000000  18420.000000  18420.000000  
mean           0.004886      3.460206    715.617101  
std            0.378271      0.690808   2569.591219  
min            0.000000      1.000000      0.000000  
25%            0.000000      3.000000      0.000000  
50%            0.000000      3.000000      0.000000  
75%            0.000000      4.000000      0.000000  
max           40.000000      5.000000  60000.000000  

[8 rows x 45 columns]

down to 18420 entries now...

checking if there are still duplicates in these our merge dataframe (repeats due to Residential building data)

In [38]:
prop_res.major_minor.value_counts()

6196600360    4
7549801930    3
7978200080    3
8586400026    3
8019200935    3
             ..
5089700760    1
3025049050    1
2473100300    1
1788800020    1
8707100015    1
Name: major_minor, Length: 18293, dtype: int64

In [39]:
prop_res[prop_res['major_minor'].isin(['6196600360', '7978200080', '1498301040'])]

excisetaxnbr major_minor   major minor documentdate  saleprice  \
290        2970236  7978200080  797820  0080   2019-01-10     435000   
291        2970236  7978200080  797820  0080   2019-01-10     435000   
292        2970236  7978200080  797820  0080   2019-01-10     435000   
820        2972255  6196600360  619660  0360   2019-01-28     995000   
821        2972255  6196600360  619660  0360   2019-01-28     995000   
822        2972255  6196600360  619660  0360   2019-01-28     995000   
823        2972255  6196600360  619660  0360   2019-01-28     995000   
9496       2999967  1498301040  149830  1040   2019-07-16    1244600   
9497       2999967  1498301040  149830  1040   2019-07-16    1244600   
9498       2999967  1498301040  149830  1040   2019-07-16    1244600   

        recordingnbr  propertytype  principaluse  saleinstrument  ...  \
290   20190115001718            11             6               3  ...   
291   20190115001718            11             6               3  ...   
292   20190115001718            11             6               3  ...   
820                             11             6               3  ...   
821                             11             6               3  ...   
822                             11             6               3  ...   
823                             11             6               3  ...   
9496  20190717001839            12             6               3  ...   
9497  20190717001839            12             6               3  ...   
9498  20190717001839            12             6               3  ...   

     fpmultistory fpfreestanding fpadditional yrbuilt  yrrenovated  \
290             0              0            0    2018            0   
291             0              0            0    1954            0   
292             0              0            0    2018            0   
820             0              0            0    1964            0   
821             0              0            0    1948         1999   
822             0              0            0    1978            0   
823             0              0            0    1956            0   
9496            0              0            0    1913            0   
9497            1              0            1    1904            0   
9498            0              0            0    1913            0   

      pcntcomplete obsolescence pcntnetcondition condition  addnlcost  
290              0            0                0         3          0  
291              0            0                0         4          0  
292              0            0                0         3          0  
820              0            0                0         4          0  
821              0            0                0         3          0  
822              0            0                0         3          0  
823              0            0                0         4          0  
9496             0            0                0         3          0  
9497             0            0                0         4          0  
9498             0            0                0         4          0  

[10 rows x 67 columns]

droppingteh duplicates again

In [40]:
prop_res.drop_duplicates(subset='major_minor', inplace=True)
prop_res.shape

(18293, 67)

Down to 18293 entries, no duplicates! just Residential homes sold in king county in 2019 (no Condos either)

bringing in parcel table and cleaning it up

In [41]:
Parcel.rename(columns=str.lower, inplace=True)
Parcel['major'] = Parcel['major'].astype(str)
Parcel['minor'] = Parcel['minor'].astype(str)
Parcel['major'] = Parcel['major'].apply(major_padded)
Parcel['minor'] = Parcel['minor'].apply(minor_padded)
Parcel['major_minor'] = Parcel['major']+Parcel['minor']
last_col = Parcel.pop(col_name)
Parcel.insert(0, col_name, last_col)

In [42]:
Parcel.columns

Index(['major_minor', 'major', 'minor', 'propname', 'platname', 'platlot',
       'platblock', 'range', 'township', 'section', 'quartersection',
       'proptype', 'area', 'subarea', 'specarea', 'specsubarea',
       'districtname', 'levycode', 'currentzoning', 'hbuasifvacant',
       'hbuasimproved', 'presentuse', 'sqftlot', 'watersystem', 'sewersystem',
       'access', 'topography', 'streetsurface', 'restrictiveszshape',
       'inadequateparking', 'pcntunusable', 'unbuildable', 'mtrainier',
       'olympics', 'cascades', 'territorial', 'seattleskyline', 'pugetsound',
       'lakewashington', 'lakesammamish', 'smalllakerivercreek', 'otherview',
       'wfntlocation', 'wfntfootage', 'wfntbank', 'wfntpoorquality',
       'wfntrestrictedaccess', 'wfntaccessrights', 'wfntproximityinfluence',
       'tidelandshoreland', 'lotdepthfactor', 'trafficnoise', 'airportnoise',
       'powerlines', 'othernuisances', 'nbrbldgsites', 'contamination',
       'dnrlease', 'adjacentgolffairway', 'adjac

In [43]:
Parcel.loc[1]

major_minor           1326069228
major                     132606
minor                       9228
propname                        
platname                     NaN
                         ...    
speciesofconcern               N
sensitiveareatract             N
waterproblems                  N
transpconcurrency              N
otherproblems                  N
Name: 1, Length: 82, dtype: object

In [44]:
prop_res = prop_res.merge(Parcel, on='major_minor', suffixes=("", "_parcel"))

In [45]:
prop_res.head()

excisetaxnbr major_minor   major minor documentdate  saleprice  \
0       2968945  4077800202  407780  0202   2019-01-01    1910000   
1       2970528  6190530110  619053  0110   2019-01-01    1089950   
2       2969165  0425400040  042540  0040   2019-01-01     412500   
3       2969387  2525059126  252505  9126   2019-01-01     541000   
4       2968880  1250203365  125020  3365   2019-01-02    1020000   

     recordingnbr  propertytype  principaluse  saleinstrument  ...  \
0  20190103000654            11             6               3  ...   
1  20190117000670            11             6               3  ...   
2  20190104000953            11             6               3  ...   
3  20190108000677            11             6               3  ...   
4  20190103000222            11             6               3  ...   

  seismichazard landslidehazard steepslopehazard stream  wetland  \
0             N               N                N      N        N   
1             N               N                N      N        N   
2             N               N                N      N        N   
3             N               N                N      N        N   
4             N               N                N      N        N   

   speciesofconcern sensitiveareatract waterproblems transpconcurrency  \
0                 N                  N             N                 N   
1                 N                  N             N                 N   
2                 N                  N             N                 N   
3                 N                  N             N                 N   
4                 N                  N             N                 N   

   otherproblems  
0              N  
1              N  
2              N  
3              N  
4              N  

[5 rows x 148 columns]

In [46]:
prop_res.shape

(18293, 148)

no loss of entries here, still at 18293!

In [47]:
prop_res['proptype'].unique()

array(['R'], dtype=object)

R - Residential, propert type included in this database only reflects residential property. That's a good sign!

In [48]:
prop_res.describe()

excisetaxnbr     saleprice  propertytype  principaluse  saleinstrument  \
count  1.829300e+04  1.829300e+04  18293.000000       18293.0    18293.000000   
mean   2.999516e+06  7.947806e+05     11.027552           6.0        3.076696   
std    1.662625e+04  5.862317e+05      0.310903           0.0        1.186828   
min    2.968844e+06  1.000000e+01     11.000000           6.0        2.000000   
25%    2.985463e+06  4.700000e+05     11.000000           6.0        3.000000   
50%    3.000295e+06  6.551010e+05     11.000000           6.0        3.000000   
75%    3.012868e+06  9.050000e+05     11.000000           6.0        3.000000   
max    3.050564e+06  1.605000e+07     18.000000           6.0       26.000000   

         salereason  propertyclass       bldgnbr  nbrlivingunits  \
count  18293.000000        18293.0  18293.000000    18293.000000   
mean       1.001859            8.0      1.003389        1.014158   
std        0.177750            0.0      0.074597        0.137401   
min        1.000000            8.0      1.000000        1.000000   
25%        1.000000            8.0      1.000000        1.000000   
50%        1.000000            8.0      1.000000        1.000000   
75%        1.000000            8.0      1.000000        1.000000   
max       18.000000            8.0      7.000000        3.000000   

            stories  ...  wfntpoorquality  wfntrestrictedaccess  \
count  18293.000000  ...     18293.000000          18293.000000   
mean       1.518778  ...         0.000547              0.005412   
std        0.534574  ...         0.023375              0.107774   
min        1.000000  ...         0.000000              0.000000   
25%        1.000000  ...         0.000000              0.000000   
50%        1.500000  ...         0.000000              0.000000   
75%        2.000000  ...         0.000000              0.000000   
max        3.500000  ...         1.000000              3.000000   

       tidelandshoreland  lotdepthfactor  trafficnoise  airportnoise  \
count       18293.000000    18293.000000  18293.000000       18293.0   
mean            0.014869        0.054010      0.199257           0.0   
std             0.170853        2.121509      0.544073           0.0   
min             0.000000        0.000000      0.000000           0.0   
25%             0.000000        0.000000      0.000000           0.0   
50%             0.000000        0.000000      0.000000           0.0   
75%             0.000000        0.000000      0.000000           0.0   
max             3.000000      105.000000      3.000000           0.0   

       nbrbldgsites  contamination  historicsite  currentusedesignation  
count  18293.000000        18293.0  18293.000000           18293.000000  
mean       0.005904            0.0      0.000929               0.003225  
std        0.106984            0.0      0.051748               0.097197  
min        0.000000            0.0      0.000000               0.000000  
25%        0.000000            0.0      0.000000               0.000000  
50%        0.000000            0.0      0.000000               0.000000  
75%        0.000000            0.0      0.000000               0.000000  
max        5.000000            0.0      3.000000               3.000000  

[8 rows x 88 columns]

total count 18293

choosing all the features of a residential property that would have a bearing on its saleprice:

**Residential building features:** (Specific to the unit)

total living square feet, year built, year renovated, bedrooms, bath(half, quarter, full), porch square feet(open, enclosed), deck, total bsement square feet, finished basement grade, daylight basement, heat system, heat source, condition

**Parcel features:** (specifi to the location of the lot)

HBU as is vacant, parking, transportation concurrency, views of Mt.Rainier, Olympics, Cascades, Territorial, Seattle skyline, Puget sound, Lake washington, Lake sammamish, small lake/river/creek, waterfront location, waterfront access rights trffic noise, airport noise, power lines, other nuisances, adjacent greenbelt.  (may want to include hazard metrics) 

In [49]:
df = prop_res.loc[:, ['saleprice', 'sqfttotliving', 'yrbuilt', 'yrrenovated', 'bedrooms',
                      'zipcode', 'sqfttotbasement', 'sqftfinbasement', 'sqftopenporch', 
                      'sqftenclosedporch', 'sqftdeck','heatsystem', 'heatsource', 'bathhalfcount', 
                      'bath3qtrcount', 'bathfullcount','condition', 'viewutilization', 'sqftgarageattached',
                      'daylightbasement', 'finbasementgrade', 'hbuasifvacant', 'inadequateparking',
                      'mtrainier', 'olympics', 'cascades', 'territorial', 'seattleskyline', 'pugetsound',
                      'lakewashington', 'lakesammamish', 'smalllakerivercreek', 'otherview', 'wfntlocation',
                      'trafficnoise', 'airportnoise', 'powerlines', 'othernuisances', 'adjacentgreenbelt']]

In [50]:
df.head()

saleprice  sqfttotliving  yrbuilt  yrrenovated  bedrooms zipcode  \
0    1910000           3990     1952         2019         6   98125   
1    1089950           3920     2018            0         5   98028   
2     412500           2280     1958            0         4   98056   
3     541000           1030     1965            0         3   98052   
4    1020000           2410     1939            0         3   98144   

   sqfttotbasement  sqftfinbasement  sqftopenporch  sqftenclosedporch  ...  \
0             1360             1360            110                  0  ...   
1              930              930            550                  0  ...   
2             1140             1140              0                  0  ...   
3                0                0            210                  0  ...   
4             1070             1070              0                  0  ...   

   lakewashington  lakesammamish  smalllakerivercreek  otherview  \
0               1              0                    0          0   
1               0              0                    0          0   
2               0              0                    0          0   
3               0              0                    0          0   
4               4              0                    0          0   

   wfntlocation  trafficnoise  airportnoise powerlines  othernuisances  \
0             0             0             0          N               N   
1             0             0             0          N               N   
2             0             1             0          N               N   
3             0             0             0          N               N   
4             0             1             0          N               N   

  adjacentgreenbelt  
0                 N  
1                 N  
2                 N  
3                 N  
4                 N  

[5 rows x 39 columns]

In [51]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 18293 entries, 0 to 18292
Data columns (total 39 columns):
 #   Column               Non-Null Count  Dtype 
---  ------               --------------  ----- 
 0   saleprice            18293 non-null  int64 
 1   sqfttotliving        18293 non-null  int64 
 2   yrbuilt              18293 non-null  int64 
 3   yrrenovated          18293 non-null  int64 
 4   bedrooms             18293 non-null  int64 
 5   zipcode              15907 non-null  object
 6   sqfttotbasement      18293 non-null  int64 
 7   sqftfinbasement      18293 non-null  int64 
 8   sqftopenporch        18293 non-null  int64 
 9   sqftenclosedporch    18293 non-null  int64 
 10  sqftdeck             18293 non-null  int64 
 11  heatsystem           18293 non-null  int64 
 12  heatsource           18293 non-null  int64 
 13  bathhalfcount        18293 non-null  int64 
 14  bath3qtrcount        18293 non-null  int64 
 15  bathfullcount        18293 non-null  int64 
 16  cond

In [130]:
df.describe()

saleprice  sqfttotliving       yrbuilt   yrrenovated      bedrooms  \
count  1.829300e+04   18293.000000  18293.000000  18293.000000  18293.000000   
mean   7.947806e+05    2149.304871   1974.732521     99.939157      3.435740   
std    5.862317e+05     962.898244     30.337347    435.816271      0.920872   
min    1.000000e+01     290.000000   1900.000000      0.000000      0.000000   
25%    4.700000e+05    1460.000000   1954.000000      0.000000      3.000000   
50%    6.551010e+05    1970.000000   1978.000000      0.000000      3.000000   
75%    9.050000e+05    2630.000000   2001.000000      0.000000      4.000000   
max    1.605000e+07   10910.000000   2020.000000   2020.000000     13.000000   

       sqfttotbasement  sqftfinbasement  sqftopenporch  sqftenclosedporch  \
count     18293.000000     18293.000000   18293.000000       18293.000000   
mean        483.548570       304.776199      72.816706           5.377303   
std         582.097386       453.810898     150.235486          36.652649   
min           0.000000         0.000000       0.000000           0.000000   
25%           0.000000         0.000000       0.000000           0.000000   
50%           0.000000         0.000000      20.000000           0.000000   
75%         960.000000       580.000000     100.000000           0.000000   
max        5060.000000      4590.000000   11570.000000        1840.000000   

           sqftdeck  ...   territorial  seattleskyline    pugetsound  \
count  18293.000000  ...  18293.000000    18293.000000  18293.000000   
mean     137.064396  ...      0.246160        0.024654      0.057508   
std      213.103559  ...      0.777602        0.262234      0.386005   
min        0.000000  ...      0.000000        0.000000      0.000000   
25%        0.000000  ...      0.000000        0.000000      0.000000   
50%        0.000000  ...      0.000000        0.000000      0.000000   
75%      220.000000  ...      0.000000        0.000000      0.000000   
max     2250.000000  ...      4.000000        4.000000      4.000000   

       lakewashington  lakesammamish  smalllakerivercreek     otherview  \
count    18293.000000   18293.000000         18293.000000  18293.000000   
mean         0.061718       0.014049             0.021976      0.014650   
std          0.408426       0.203752             0.256041      0.189674   
min          0.000000       0.000000             0.000000      0.000000   
25%          0.000000       0.000000             0.000000      0.000000   
50%          0.000000       0.000000             0.000000      0.000000   
75%          0.000000       0.000000             0.000000      0.000000   
max          4.000000       4.000000             4.000000      4.000000   

       wfntlocation  trafficnoise  airportnoise  
count  18293.000000  18293.000000       18293.0  
mean       0.105286      0.199257           0.0  
std        0.863694      0.544073           0.0  
min        0.000000      0.000000           0.0  
25%        0.000000      0.000000           0.0  
50%        0.000000      0.000000           0.0  
75%        0.000000      0.000000           0.0  
max        9.000000      3.000000           0.0  

[8 rows x 33 columns]

In [131]:
df.corr()

saleprice  sqfttotliving   yrbuilt  yrrenovated  \
saleprice             1.000000       0.629960  0.083728     0.116650   
sqfttotliving         0.629960       1.000000  0.335873     0.052039   
yrbuilt               0.083728       0.335873  1.000000    -0.252578   
yrrenovated           0.116650       0.052039 -0.252578     1.000000   
bedrooms              0.279942       0.618696  0.200411     0.016790   
sqfttotbasement       0.274627       0.309884 -0.224629     0.070194   
sqftfinbasement       0.293731       0.415486 -0.114049     0.087986   
sqftopenporch         0.273600       0.357653  0.150373     0.020346   
sqftenclosedporch     0.007676      -0.003029 -0.082272     0.019971   
sqftdeck              0.212555       0.244592  0.044519     0.057851   
heatsystem            0.137244       0.227525  0.178416    -0.001193   
heatsource           -0.052527      -0.029455  0.153384     0.038291   
bathhalfcount         0.218909       0.355068  0.390476    -0.054996   
bath3qtrcount         0.234477       0.263977  0.048808     0.048519   
bathfullcount         0.325288       0.568801  0.393462     0.038133   
condition            -0.043637      -0.106696 -0.342769    -0.090368   
sqftgarageattached    0.171776       0.456309  0.427590    -0.088992   
finbasementgrade      0.235494       0.241750 -0.037948     0.052852   
hbuasifvacant        -0.009956      -0.036264 -0.086453     0.008686   
inadequateparking     0.049376       0.022528  0.033775     0.014924   
mtrainier             0.076261       0.065160 -0.005720     0.040119   
olympics              0.154945       0.126138 -0.045072     0.053250   
cascades              0.125105       0.131224 -0.026904     0.041741   
territorial           0.312533       0.240496 -0.059007     0.096305   
seattleskyline        0.181434       0.099264 -0.020962     0.036883   
pugetsound            0.106470       0.088531 -0.067391     0.075409   
lakewashington        0.375150       0.174054 -0.044099     0.070361   
lakesammamish         0.158958       0.097847  0.015510     0.029599   
smalllakerivercreek   0.041940       0.030563 -0.006971     0.018536   
otherview             0.098647       0.085242  0.025544    -0.001783   
wfntlocation          0.259750       0.092764 -0.007644     0.050285   
trafficnoise         -0.001479      -0.056715 -0.097873     0.037215   
airportnoise               NaN            NaN       NaN          NaN   

                     bedrooms  sqfttotbasement  sqftfinbasement  \
saleprice            0.279942         0.274627         0.293731   
sqfttotliving        0.618696         0.309884         0.415486   
yrbuilt              0.200411        -0.224629        -0.114049   
yrrenovated          0.016790         0.070194         0.087986   
bedrooms             1.000000         0.220708         0.296155   
sqfttotbasement      0.220708         1.000000         0.853537   
sqftfinbasement      0.296155         0.853537         1.000000   
sqftopenporch        0.174940        -0.005648         0.014079   
sqftenclosedporch   -0.017515         0.014929         0.007904   
sqftdeck             0.100441         0.274132         0.296975   
heatsystem           0.168008         0.095200         0.101414   
heatsource          -0.046377        -0.098577        -0.020566   
bathhalfcount        0.177216        -0.057833        -0.023241   
bath3qtrcount        0.215472         0.285761         0.313822   
bathfullcount        0.438646         0.055916         0.126058   
condition           -0.000161         0.094291         0.136141   
sqftgarageattached   0.280272        -0.331527        -0.165987   
finbasementgrade     0.161556         0.804652         0.826562   
hbuasifvacant       -0.011077         0.021211        -0.001844   
inadequateparking    0.000404        -0.007173         0.000701   
mtrainier            0.015546         0.042309         0.052209   
olympics             0.024748         0.162034         0.163403   
cascades             0.045932  

In [52]:
df.to_csv('../data/our_dataframe.csv')

In [53]:
prop_res.to_csv('../data/df_with_all_columns.csv')